allensdk/brain_observatory/behavior/data_objects/metadata/ophys_experiment_metadata/ophys_project_code.py

```diff
@@ -0,0 +1,29 @@
+from allensdk.brain_observatory.behavior.data_objects.metadata.\
+    behavior_metadata.project_code import ProjectCode
+from allensdk.internal.api import PostgresQueryMixin
+
+
+class OphysProjectCode(ProjectCode):
+    """Unique identifier for the project this OphysExperiment is associated
+    with. Project ids can be used internally to extract a project code.
+
+    If the returned project id is null, return -1 for the id.
+    """
+
+    @classmethod
+    def from_lims(cls, ophys_experiment_id: int,
+                  lims_db: PostgresQueryMixin) -> "OphysProjectCode":
+        query = f"""
```

Comment:
This class is confusing since there should be a single source of truth for the project code. An ophys session is always associated with a behavior session. Is the project code always populated in the behavior sessions table? Can we also just look it up from there?

Reply:
Mostly being cautions. OphysSessions seem to always have a project associated with them. However, it seems that these projects are changed in some cases so consider this a catch for before a project_id is assigned.